teb_local_planner/src/teb_local_planner_ros.cpp

```diff
@@ -1041,30 +1055,44 @@ void TebLocalPlannerROS::configureBackupModes(std::vector<geometry_msgs::msg::Po
             RCLCPP_INFO(logger_, "TebLocalPlannerROS: oscillation recovery disabled/expired.");
         }
     }
-
 }
-     
 
-void TebLocalPlannerROS::setSpeedLimit(
-    const double & speed_limit, const bool & percentage)
-{
+void TebLocalPlannerROS::setSpeedLimit(const double& speed_limit,
+                                       const bool& percentage) {
   if (speed_limit == nav2_costmap_2d::NO_SPEED_LIMIT) {
-    // Restore default value
+    // Restore Defaults
     cfg_->robot.max_vel_x = cfg_->robot.base_max_vel_x;
-    cfg_->robot.base_max_vel_x_backwards = cfg_->robot.base_max_vel_x_backwards;
-    cfg_->robot.base_max_vel_y = cfg_->robot.base_max_vel_y;
-    cfg_->robot.base_max_vel_theta = cfg_->robot.base_max_vel_theta;
+    cfg_->robot.max_vel_x_backwards = cfg_->robot.base_max_vel_x_backwards;
+    cfg_->robot.max_vel_y = cfg_->robot.base_max_vel_y;
+    cfg_->robot.max_vel_theta = cfg_->robot.base_max_vel_theta;
   } else {
     if (percentage) {
-      // Speed limit is expressed in % from maximum speed of robot
-      cfg_->robot.max_vel_x = cfg_->robot.base_max_vel_x * speed_limit / 100.0;
-      cfg_->robot.base_max_vel_x_backwards = cfg_->robot.base_max_vel_x_backwards * speed_limit / 100.0;
-      cfg_->robot.base_max_vel_y = cfg_->robot.base_max_vel_y * speed_limit / 100.0;
-      cfg_->robot.base_max_vel_theta = cfg_->robot.base_max_vel_theta * speed_limit / 100.0;
+      if (speed_limit > 1.0 || speed_limit < 0.0) {
+        RCLCPP_WARN_THROTTLE(
+            logger_, *(clock_), 100,
+            "Percentage given outside the range 0-1. Using base velocities.");
+        // Restore defaults
+        cfg_->robot.max_vel_x = cfg_->robot.base_max_vel_x;
+        cfg_->robot.max_vel_x_backwards = cfg_->robot.base_max_vel_x_backwards;
+        cfg_->robot.max_vel_y = cfg_->robot.base_max_vel_y;
+        cfg_->robot.max_vel_theta = cfg_->robot.base_max_vel_theta;
+      } else {
+        // Speed limit is expressed in % from maximum speed of robot
+        cfg_->robot.max_vel_x =
+            cfg_->robot.base_max_vel_x * speed_limit / 100.0;
+        cfg_->robot.max_vel_x_backwards =
+            cfg_->robot.base_max_vel_x_backwards * speed_limit / 100.0;
+        cfg_->robot.max_vel_y =
+            cfg_->robot.base_max_vel_y * speed_limit / 100.0;
+        cfg_->robot.max_vel_theta =
+            cfg_->robot.base_max_vel_theta * speed_limit / 100.0;
+      }
```

Comment:
Let's ping them about the open PR

Reply:
I was thinking about it but it looks like no one is responsible. See [here](https://github.com/rst-tu-dortmund/teb_local_planner/issues/373#issuecomment-1226549953). Will leave a message on the PR nevertheless

Reply:
mmm one PR was merged 2 weeks ago